packages/backend-modules/republik-crowdfundings/lib/generateMemberships.js

```diff
@@ -183,15 +189,18 @@ module.exports = async (pledgeId, pgdb, t, redis) => {
               membership,
             }
           } else {
-            // Cancel active memberships because bought package (option) contains
-            // a better abo.
-            if (MONTHLY_ABO_UPGRADE_PKGS.includes(pkg.name)) {
+            // Cancel active membership(s) if there is an upgrade path available
+            const membershipTypes = UPGRADE_PKG_PATHS.filter(
+              ({ toPackageNames }) => toPackageNames.includes(pkg.name),
+            ).map(({ fromMembershipType }) => fromMembershipType)
+
+            if (membershipTypes.length) {
               cancelableMemberships = activeMemberships.filter(
-                (m) => m.name === 'MONTHLY_ABO' && m.renew === true,
+                (m) => membershipTypes.includes(m.name) && m.renew === true,
```

Comment:
Couldn't that be left out, if all membershipTypes are  cancelable?
```suggestion
               (m) => m.renew === true,
```

Reply:
This code change is somewhat obsolete and might have to roll back. (Changed logic on how to handel `YEARLY_ABO` quite a bit.)

Reply:
I'll stick to this after all.

While all membership are indeed cancellable, `MONTHLY_ABO` and – new – `YEARLY_ABO` require membership to cancel after you bought a new membership.

`UPGRADE_PKG_PATHS` discribes `from...` which current type `to...` which type a cancellation is necessary.

`YEARLY_ABO` behaves much like `MONTHLY_ABO`: It will show user a "upgrade link" on account page as long as it is not cancelled.